File: pset1/mario/less/mario.c
#include <stdio.h>
#include <string.h>
#include <stdlib.h>
/* Recreates this half-pyramid using hashes from Nintendo’s Super Mario Brothers */


int main(void)
{
    char *end;
    char buffer[5];
    int num;

    /* Gets an integer between 0 and 23*/
    do
    {
        printf("Height: ");
        if (!fgets(buffer, sizeof buffer, stdin))
        {
            break;
        }
        buffer[strlen(buffer) - 1] = 0;
        num = strtol(buffer, &end, 10);
    }
    while (end != buffer + strlen(buffer) || num < 1 || num > 8 || strlen(buffer) == 0);

    /* Displays the half Pyramid */
    for (int x = 1; x <= num ; x++)
    {
        for (int y = num; y > 0 ; y--)
        {
            if (y <= x)
            {
                printf("#");
            }
            else
            {
                printf(" ");
            }
        }
        printf("\n");
    }
    return (0);
}

File: pset6/mario/more/mario.py
# Recreates this half-pyramid using hashes from Nintendo’s Super Mario Brothers
import cs50
# Gets an integer between 0 and 8
while True:
    try:
        height = cs50.get_int("Height: ")
        if 1 <= height <= 8:
            break
    except ValueError:
        continue

# Displays the half Pyramid
for x in range(1, height+1):
    for y in range(height, 0, -1):
        if (y <= x):
            print("#", end='')
        else:
            print(" ", end='')
    print("  ", end='')
    for y in range(0, height, 1):
        if (y < x):
            print("#", end='')
    print("")